packages/lab/src/tabs-next/TabstripNext.tsx

```diff
@@ -0,0 +1,177 @@
+import { makePrefixer, useControlled, useId } from "@salt-ds/core";
+import clsx from "clsx";
+import {
+  Children,
+  cloneElement,
+  isValidElement,
+  PropsWithChildren,
+  ReactNode,
+  useCallback,
+  useRef,
+} from "react";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { Overflow, OverflowItem } from "@fluentui/react-overflow";
+import { TabNext } from "./TabNext";
+import { TabElement, TabProps } from "./TabsNextTypes";
+import { OverflowMenu } from "./OverflowMenu";
+import tabstripCss from "./TabstripNext.css";
+
+const noop = () => undefined;
+
+const withBaseName = makePrefixer("saltTabstripNext");
+
+function isTab(child: ReactNode | TabElement): child is TabElement {
+  return isValidElement(child) && child.type === TabNext;
+}
+
+export type TabstripNextProps = PropsWithChildren<{
+  /* Value for the uncontrolled version. Set to null in order to have no tabs be selected. */
+  activeTabIndex?: number | null;
+  /* Callback for the controlled version. */
+  onActiveChange?: (index?: number) => void;
+  /* Initial value for the uncontrolled version. Set to null in order to have no tabs be selected by default. */
+  defaultActiveTabIndex?: number | null;
+  /* Align the tabs to the center. Left aligned by default. */
+  align?: "center";
+  /* Set a tab max-width in order to enable tab truncation */
+  tabMaxWidth?: number;
+  /* 
+  Function to generate a unique id for each tab. Necessary for overflow to work. 
+  By default it uses the label, therefore the Tab label has to be unique. */
+  getTabId?: (label: string) => string;
+}>;
+
+export const TabstripNext = ({
+  children,
+  activeTabIndex: activeTabIndexProp,
+  defaultActiveTabIndex,
+  onActiveChange,
+  align,
+  tabMaxWidth,
+  getTabId: getTabIdProp,
+}: TabstripNextProps) => {
+  const targetWindow = useWindow();
+  useComponentCssInjection({
+    testId: "salt-tabstrip-next",
+    css: tabstripCss,
+    window: targetWindow,
+  });
+  const uniqueId = useId();
+  const _getTabId = useCallback(
+    (label: string) => {
+      return `tab-${uniqueId ?? "unknown"}-${label}`;
+    },
+    [uniqueId]
+  );
+  const getTabId = getTabIdProp || _getTabId;
+  const tabs = Children.toArray(children)
+    .filter(isTab)
+    .map((tab) => {
+      const { label } = tab.props;
+      const id = tab.props.id || getTabId(label);
+      return { tab, id };
+    });
```

Comment:
there's a lot of hard work to create these tab `id`s (also in Tabs). could it be possible to create the tab id in Tab?

Reply:
No, because we need these ids in Tabstrip to pass them to overflow menu and overflow library